app/src/ui/lib/ui-activity-monitor.ts

```diff
@@ -0,0 +1,138 @@
+import { Emitter, Disposable } from 'event-kit'
+import { ipcRenderer } from 'electron'
+
+/**
+ * Describes the interface of a UI activity monitor.
+ *
+ * We define a UI activity monitor as an object which can
+ * be used to be notified whenever the user interacts with the
+ * application.
+ */
+export interface IUiActivityMonitor {
+  /**
+   * Subscribe to get notified when the user interacts
+   * with the application. The first, and only argument
+   * to the event handler will be a value indicating the
+   * kind of action detected (mouse/pointer, keyboard etc).
+   *
+   * @return A disposable object which, when disposed will
+   *         terminate the subscription and prevent any
+   *         further calls to the handler.
+   */
+  onActivity(handler: (kind: UiActivityKind) => void): Disposable
+}
+
+/**
+ * A value indicating a user activity type (mouse/pointer, keyboard etc).
+ */
+export type UiActivityKind = 'pointer' | 'keyboard' | 'menu'
+
+/**
+ * UI Activity monitor for user interactions within GitHub Desktop.
+ */
+export class UiActivityMonitor implements IUiActivityMonitor {
+  private readonly emitter = new Emitter()
+  private subscriberCount = 0
+
+  /**
+   * Subscribe to get notified when the user interacts
+   * with the application. The first, and only argument
+   * to the event handler will be a value indicating the
+   * kind of action detected (mouse/pointer, keyboard etc).
+   *
+   * @return A disposable object which, when disposed will
+   *         terminate the subscription and prevent any
+   *         further calls to the handler.
+   */
+  public onActivity(handler: (kind: UiActivityKind) => void): Disposable {
+    const emitterDisposable = this.emitter.on('activity', handler)
+
+    if (this.subscriberCount++ === 0) {
```

Comment:
Can the this be changed to

```ts
if (this.subscriberCount === 0) {
  this.subscriberCount++
  this.startTracking()
}
```

to keep state changing operators separate from Boolean expressions.

Reply:
It can, and probably should. Pre-increment and post-increment are not the epitome of easily understandable code.